### PR TITLE
Add FFmpegSlice postprocessor

### PR DESCRIPTION
**Pull request by marcwebbie**
## Please follow the guide below
- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your _pull request_ (like that [x])
- Use _Preview_ tab to see how your _pull request_ will actually look like

---
### Before submitting a _pull request_ make sure you have:
- [X] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [X] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [X] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)
### What is the purpose of your _pull request_?
- [ ] Bug fix
- [ ] Improvement
- [ ] New extractor
- [X] New feature

---
### Description of your _pull request_ and other information

Explanation of your _pull request_ in arbitrary form goes here. Please make sure the description explains the purpose and effect of your _pull request_ and is worded well enough to be understood. Provide as much context and examples as possible.

Gives the possibility of slicing downloaded media. The command below will download video and cut a slice from second 3 to second 8. When finished processng It will generate a 5 seconds long clip as a result.

```
youtube-dl https://www.youtube.com/watch?v=gwWzuHPoq7E --slice-start 3 --slice-end 8
```

### CODE REVIEW

File: youtube_dl/options.py [hunk 1]
```diff
@@ -757,6 +757,14 @@ def _scrub_eq(o):
         metavar='FORMAT', dest='recodevideo', default=None,
         help='Encode the video to another format if necessary (currently supported: mp4|flv|ogg|webm|mkv|avi)')
     postproc.add_option(
+        '--slice-start',
+        metavar='START_TIME', dest='slicestart', default=None,
```
Comment (dstftw) [inline youtube_dl/options.py]: `metavar` should be `TIME`, it should also support various formats (use `parse_duration`).

File: youtube_dl/postprocessor/ffmpeg.py [hunk 2]
```diff
@@ -577,3 +577,25 @@ def run(self, info):
...
+    def __init__(self, downloader, slice_start_time, slice_end_time):
+        super(FFmpegSlicePP, self).__init__(downloader)
+        self.slice_start_time = slice_start_time
+        self.slice_end_time = slice_end_time
+
+    def run(self, info):
+        filename = info['filepath']
+        temp_filename = prepend_extension(filename, 'temp')
+        start_time = self.slice_start_time or "0"
+        end_time = self.slice_end_time or "{}".format(info['duration'])
```
Comment (dstftw) [inline youtube_dl/postprocessor/ffmpeg.py]: `{}` does not work in python 2.6. Use single quotes consistently.

File: youtube_dl/options.py (hunk 1 above)
Comment (marcwebbie) [inline youtube_dl/options.py]: 👍

File: youtube_dl/postprocessor/ffmpeg.py (hunk 2 above)
Comment (marcwebbie) [inline youtube_dl/postprocessor/ffmpeg.py]: 👍

**Comment by marcwebbie** [comment]
Hello @dstftw It seems that the build is breaking due to extractors, couldn't see any failing tests related to my code in Travis. How should I proceed here?

**Comment by cribeiro** [comment]
Any news on this pull request?

**Comment by remitamine** [comment]
isn't it better add support for this feature at the downloader level instead of a post prossesor.
for the other protocols not supported by ffmpeg(dash, f4m), it won't be too difficult to add support for this options.
using the method in this PR, i see that it's a waste to download the full video while you're interested in a small part of the video.

**Comment by mpenkov** [comment]
It's been a year since the last update. What should we do with this PR?

**Pull request closed without merging**